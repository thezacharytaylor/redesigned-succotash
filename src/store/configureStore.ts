import { configureStore } from '@reduxjs/toolkit';
import { createReducer } from './reducers';

export function configureAppStore() {
  const store = configureStore({
    reducer: createReducer(),
    middleware: defaultMiddleware => [...defaultMiddleware()],
    devTools:
      /* istanbul ignore next line */
      process.env.NODE_ENV !== 'production' ||
      process.env.PUBLIC_URL.length > 0,
  });

  return store;
}
