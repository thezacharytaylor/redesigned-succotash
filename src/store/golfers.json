[
  {
    "id": 1,
    "name": "Ken Price",
    "score": 18,
    "date": "10/24/2022",
    "inCup": true,
    "qualified": true,
    "checkedIn": true,
    "seed": 1,
    "previousPlayer": false
  },
  {
    "id": 2,
    "name": "David Green",
    "score": 19,
    "date": "10/24/2022",
    "inCup": true,
    "qualified": true,
    "checkedIn": true,
    "seed": 2,
    "previousPlayer": true
  },
  {
    "id": 3,
    "name": "Carl Jones",
    "score": 9,
    "date": "11/24/2022",
    "inCup": false,
    "qualified": false,
    "checkedIn": false,
    "seed": 3,
    "previousPlayer": false
  }
]
