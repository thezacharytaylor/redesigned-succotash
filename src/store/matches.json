[
  {
    "id": 20221,
    "title":
  }
]
