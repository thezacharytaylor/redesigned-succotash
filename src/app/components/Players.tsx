import { useEffect, useState } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { setName, setScore } from 'features/player-input/inputSlice';
import { RootState } from 'store/reducers';
import LeaderBoard from './Leaderboard';
import Form from './Form';
import dayjs from 'dayjs';
import PlayerData from '../../store/golfers.json';

interface DefaultPlayer {
  id: number;
  name: string;
  score: number;
  date: string;
  inCup: boolean;
  qualified: boolean;
  checkedIn: boolean;
  seed: number;
  previousPlayer: boolean;
}

const defaultPlayer: DefaultPlayer = {
  id: -1,
  name: '',
  score: 0,
  date: '',
  inCup: false,
  qualified: false,
  checkedIn: false,
  seed: 16,
  previousPlayer: false,
};

const Players = () => {
  const playerInput = useSelector((state: RootState) => state.playerInput);
  const dispatch = useDispatch();
  const [players, setPlayers] = useState([defaultPlayer]);

  useEffect(() => {
    setPlayers([...PlayerData]);
  }, []);

  // TODO: Move to form when players moved to store
  // TODO: Make Slice
  // TODO: Make/Add reducer
  // TODO: Cleanup code
  const handleFormSubmit = (event: { preventDefault: () => void }) => {
    event.preventDefault();
    const updatedPlayer = updatePlayer();

    if (!updatedPlayer) {
      addPlayer();
    }
  };

  // TODO: To Store
  const updatePlayer = () => {
    let locatedPlayer = false;
    const updatedPlayers = players.map(golfer => {
      if (golfer.name === playerInput.name) {
        golfer.score = playerInput.score;
        golfer.date = dayjs().format('MM/DD/YYYY');
        locatedPlayer = true;
      }

      return golfer;
    });

    setPlayers(updatedPlayers);
    return locatedPlayer;
  };

  // TODO: To Store
  const addPlayer = () => {
    setPlayers([
      ...players,
      {
        id: players[players.length - 1].id + 1,
        name: playerInput.name,
        score: +playerInput.score,
        date: dayjs().format('MM/DD/YYYY'),
        inCup: true,
        qualified: true,
        checkedIn: true,
        seed: 2,
        previousPlayer: false,
      },
    ]);

    dispatch(setName(''));
    dispatch(setScore(0));
  };

  // To Store
  // const deletePlayer = id => {};

  return (
    <div className="p-10 my-5 border-solid rounded-md shadow-sm bg-primary-content border-slate-50">
      <LeaderBoard
        headings={['Rank', 'Name', 'Score', 'Date']}
        players={players}
      />
      <hr className="my-4 border-solid border-secondary-content" />
      <h2 className="mb-4 text-lg font-bold text-gray-800">Add Score</h2>
      <Form submit={handleFormSubmit} players={players} />
    </div>
  );
};

export default Players;
