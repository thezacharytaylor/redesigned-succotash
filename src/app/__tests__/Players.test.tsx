import renderer from 'react-test-renderer';
import userEvent from '@testing-library/user-event';
import Players from 'app/components/Players';
import { Provider } from 'react-redux';
import { HelmetProvider } from 'react-helmet-async';
import { configureAppStore } from 'store/configureStore';

const store = configureAppStore();
const user = userEvent.setup();

describe('Load app', () => {
  test('should render and match the snapshot', () => {
    const domTree = renderer
      .create(
        <Provider store={store}>
          <HelmetProvider>
            <Players />
          </HelmetProvider>
        </Provider>,
      )
      .toJSON();
    expect(domTree).toMatchSnapshot(); //
  });
});

describe('Add player', () => {
  test('Should add fake player to leaderboard', () => {
    const domTree = renderer
      .create(
        <Provider store={store}>
          <HelmetProvider>
            <Players />
          </HelmetProvider>
        </Provider>,
      )
      .toJSON();
  });
});
