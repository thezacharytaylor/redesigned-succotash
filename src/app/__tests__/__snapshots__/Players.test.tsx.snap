// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Load app should render and match the snapshot 1`] = `
<div
  className="p-10 my-5 bg-gray-200 border-solid rounded-md shadow-sm border-slate-50"
>
  <h2
    className="mb-4 text-lg font-bold text-gray-800"
    id="leaderboard-header"
  >
      Individual Qualifying High Scores
  </h2>
  <div
    className="overflow-x-auto"
  >
    <table
      aria-labelledby="leaderboard-header"
      className="table w-full table-zebra"
      role="grid"
    >
      <thead>
        <tr>
          <th
            className=" !bg-green-900 text-white"
            scope="col"
          >
            Rank
          </th>
          <th
            className="text-left !bg-green-900 text-white"
            scope="col"
          >
            Name
          </th>
          <th
            className=" !bg-green-900 text-white"
            scope="col"
          >
            Score
          </th>
          <th
            className=" !bg-green-900 text-white"
            scope="col"
          >
            Date
          </th>
        </tr>
      </thead>
      <tbody>
        <tr>
          <td
            className="pl-6 !bg-green-700"
          >
            <div
              className="p-0 text-white bg-transparent border-none cursor-default"
              role="note"
              tabIndex={0}
            >
              1
            </div>
          </td>
          <td
            className=" !bg-green-700"
          >
            <div
              className="p-0 text-white bg-transparent border-none cursor-default"
              role="note"
              tabIndex={0}
            />
          </td>
          <td
            className=" !bg-green-700"
          >
            <div
              className="p-0 text-white bg-transparent border-none cursor-default"
              role="note"
              tabIndex={0}
            >
              0
            </div>
          </td>
          <td
            className=" !bg-green-700"
          >
            <div
              className="p-0 text-white bg-transparent border-none cursor-default"
              role="note"
              tabIndex={0}
            />
          </td>
        </tr>
      </tbody>
    </table>
  </div>
  <hr
    className="my-4 border-gray-800 border-solid"
  />
  <h2
    className="mb-4 text-lg font-bold text-gray-800"
  >
    Add Score
  </h2>
  <form
    action="#"
    className="relative flex input-group"
    onSubmit={[Function]}
  >
    <input
      className="input input-bordered w-full max-w-xs false"
      name="name"
      onChange={[Function]}
      placeholder="Name"
      type="text"
      value=""
    />
    <div
      className="absolute bottom-0 w-full"
    />
    <input
      className="input input-bordered w-full max-w-xs max-w-[4rem]"
      name="score"
      onChange={[Function]}
      placeholder={1}
      type="text"
      value={0}
    />
    <button
      className="px-4 text-lg text-white bg-gray-700 lg:text-2xl btn"
      type="submit"
    >
      +
    </button>
  </form>
</div>
`;

exports[`should render and match the snapshot 1`] = `
<div
  className="p-10 my-5 bg-gray-200 border-solid rounded-md shadow-sm border-slate-50"
>
  <h2
    className="mb-4 text-lg font-bold text-gray-800"
    id="leaderboard-header"
  >
      Individual Qualifying High Scores
  </h2>
  <div
    className="overflow-x-auto"
  >
    <table
      aria-labelledby="leaderboard-header"
      className="table w-full table-zebra"
      role="grid"
    >
      <thead>
        <tr>
          <th
            className=" !bg-green-900 text-white"
            scope="col"
          >
            Rank
          </th>
          <th
            className="text-left !bg-green-900 text-white"
            scope="col"
          >
            Name
          </th>
          <th
            className=" !bg-green-900 text-white"
            scope="col"
          >
            Score
          </th>
          <th
            className=" !bg-green-900 text-white"
            scope="col"
          >
            Date
          </th>
        </tr>
      </thead>
      <tbody>
        <tr>
          <td
            className="pl-6 !bg-green-700"
          >
            <div
              className="p-0 text-white bg-transparent border-none cursor-default"
              role="note"
              tabIndex={0}
            >
              1
            </div>
          </td>
          <td
            className=" !bg-green-700"
          >
            <div
              className="p-0 text-white bg-transparent border-none cursor-default"
              role="note"
              tabIndex={0}
            />
          </td>
          <td
            className=" !bg-green-700"
          >
            <div
              className="p-0 text-white bg-transparent border-none cursor-default"
              role="note"
              tabIndex={0}
            >
              0
            </div>
          </td>
          <td
            className=" !bg-green-700"
          >
            <div
              className="p-0 text-white bg-transparent border-none cursor-default"
              role="note"
              tabIndex={0}
            />
          </td>
        </tr>
      </tbody>
    </table>
  </div>
  <hr
    className="my-4 border-gray-800 border-solid"
  />
  <h2
    className="mb-4 text-lg font-bold text-gray-800"
  >
    Add Score
  </h2>
  <form
    action="#"
    className="relative flex input-group"
    onSubmit={[Function]}
  >
    <input
      className="input input-bordered w-full max-w-xs false"
      name="name"
      onChange={[Function]}
      placeholder="Name"
      type="text"
      value=""
    />
    <div
      className="absolute bottom-0 w-full"
    />
    <input
      className="input input-bordered w-full max-w-xs max-w-[4rem]"
      name="score"
      onChange={[Function]}
      placeholder={1}
      type="text"
      value={0}
    />
    <button
      className="px-4 text-lg text-white bg-gray-700 lg:text-2xl btn"
      type="submit"
    >
      +
    </button>
  </form>
</div>
`;
